src/main/java/com/google/maps/GeolocationApi.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.google.maps;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.maps.errors.ApiError;
+import com.google.maps.errors.ApiException;
+import com.google.maps.internal.ApiConfig;
+import com.google.maps.internal.ApiResponse;
+import com.google.maps.model.CellTower;
+import com.google.maps.model.GeolocationPostPayload;
+import com.google.maps.model.GeolocationResult;
+import com.google.maps.model.WifiAccessPoint;
+
+/*
+ *  The Google Maps Geolocation API returns a location and accuracy radius based on information
+ *  about cell towers and WiFi nodes that the mobile client can detect.
+ *
+ *  https://developers.google.com/maps/documentation/geolocation/intro#top_of_page
+ */
+public class GeolocationApi {
+  static final String API_BASE_URL = "https://www.googleapis.com";
+
+  static final ApiConfig GEOLOCATION_API_CONFIG = new ApiConfig("/geolocation/v1/geolocate")
+      .hostName(API_BASE_URL)
+      .supportsClientId(false)
+      .fieldNamingPolicy(FieldNamingPolicy.IDENTITY);
+
+  private GeolocationApi () {
+  }
+
+  public static PendingResult<GeolocationResult> geolocate(GeoApiContext context, GeolocationPostPayload payload) {
```

Comment:
Rather than forcing developers to create a POJO and set each of the fields themselves, this library prefers to structure the API call method in a builder fashion.

e.g. in the Directions API we do this:

    DirectionsResult result = DirectionsApi.newRequest(context)
        .mode(TravelMode.BICYCLING)
        .avoid(RouteRestriction.HIGHWAYS, RouteRestriction.TOLLS, RouteRestriction.FERRIES)
        .units(Unit.METRIC)
        .region("au")
        .origin("Sydney")
        .destination("Melbourne").await();

This saves users from having to create a separate object, allows for brevity for simple requests, and enables us to create an immutable payload (important for thread safety).

Can you please update this to follow the library's existing style? You'll likely need to introduce a GeolocationApiRequest object, but you should be able to model it on the similar classes that exist.